docs/userguide.rst

```diff
@@ -252,9 +252,31 @@ option to the ``reana-client start`` or ``reana-client run`` commands.
 
 .. code-block:: console
 
-   $ reana-client start -w workflow.1 -o target='gendata'
+   $ reana-client start -w myanalysis.42 -o TARGET='gendata'
    # or
-   $ reana-client run -w workflow.1 -o target='gendata'
+   $ reana-client run -w myanalysis.42 -o TARGET='gendata'
+
+If you want to start workflow execution from a certain step you should use
+operational option ``FROM`` with a desired step name.
+
+.. code-block:: console
+
+   $ reana-client start -w myanalysis.42 -o FROM='fitdata'
+   # or
+   $ reana-client run -w myanalysis.42 -o FROM='fitdata'
+
+If you want to restart workflow on the same workspace you should use
+``--restart`` flag of ``reana-client start``.
+
+Note that workflow restarting can be used in a combination with operational
+options ``FROM`` and ``TARGET``.
+
+.. code-block:: console
+
+   # Restarts workflow on the same workspace from step named fitdata
+   $ reana-client start -w myanalysis.42 -o FROM='fitdata' --restart
```

Comment:
We don't need single quotes in CLI command invokations, so we can remove them from the examples, so that they would read more nicely:

`-o TARGET=gendata`
`-o FROM=gendata`
`...`